### ISSUE TITLE
Settings.stringSet(key: String? = null, defaultValue: Set<String> = emptySet(), separator = ",")

### ISSUE THREAD

**Issue by vanniktech** [russhwolf/multiplatform-settings#67]
Implementation could look something like this:

```
internal class SettingsDelegateSetString(
  private val settings: Settings,
  private val key: String,
  private val default: Set<String>,
  private val separator: String
) {
  operator fun getValue(thisRef: Any?, prop: KProperty<*>) =
    settings.getStringOrNull(key)
      ?.split(separator)
      ?.toSet()
      ?: default

  operator fun setValue(thisRef: Any?, prop: KProperty<*>, value: Set<String>) =
    settings.putString(key, value.joinToString(separator = separator))
}
```

Do you think that's a good fit?

**Comment by russhwolf** [comment]
Is it useful? I've never found much need for the StringSet APIs in SharedPreferences.

The solution I find more interesting for non-primitive data is the kotlinx.serialization integration I have prototyped in #36. This would generalize much better to other simple datatypes without needing to special-case anything. Looking to probably release that in the near future as part of 0.7. Do you think that will fit your use-case? You'd be able to do something like 
```
val stringSet by settings.serializationDelegate(SetSerializer(String.serializer()), "key")
```

**Comment by vanniktech** [comment]
Yup that would work too.

**Comment by russhwolf** [comment]
The serialization integration is now released in 0.7 so I'm closing this.
